nebula_decoders/src/nebula_decoders_hesai/decoders/pandar_at_decoder.cpp

```diff
@@ -58,13 +51,15 @@ PandarATDecoder::PandarATDecoder(
 
   scan_pc_.reset(new NebulaPointCloud);
   overflow_pc_.reset(new NebulaPointCloud);
+  scan_pc_->reserve(1200 * 128);
```

Comment:
Can you please convert these numbers to `constexpr` in the header and assign them a name?

Reply:
Maybe it is better to get rid of the reserve altogether:
The 1200*128 case is only valid for single return and would be double for dual return.
Thus, we could either:
* Reserve space for dual return (wasteful if only single return is used)
* Reserve space for single return (vector would grow on first dual return scan)
* Reserve nothing (vector grows on first scan)

I am in favor of the last option as it is the most concise and does not decrease performance except for the first scan.

Reply:
> Is Nebula supporting the switching of return modes on the fly?

It should be supported inherently but not tested.
I would be in favor of option one as well.

Reply:
Done :+1: 
There are now constants for scan resolution (width * LASER_NUM) and MAX_RETURNS which are then used to reserve space for the maximum number of points that can occur.